src/main/java/nurseybook/model/task/GhostTask.java

```diff
@@ -1,65 +1,184 @@
+        DateTime copyDt = new DateTime(getDateTime().getStringDate(), getDateTime().getStringTime());
+        Set<Name> copyRelatedNames = new HashSet<>();
+        for (Name name : getRelatedNames()) {
+            copyRelatedNames.add(new Name(name.fullName));
+        }
+        Status copyStatus = new Status(getStatus().isDone, getStatus().isOverdue);
+        Recurrence copyRecurrence = new Recurrence(getRecurrence().toString());
+
+        return new GhostTask(copyDesc, copyDt, copyRelatedNames, copyStatus, copyRecurrence);
+    }
 
     /**
-     * Constructs a {@code GhostTask}.
+     * Marks task as done.
      *
-     * @param isGhostTask True if GhostTask; false otherwise.
+     * @return same task object that has been marked as done
      */
-    public GhostTask(String isGhostTask) {
-        requireNonNull(isGhostTask);
-        checkArgument(isValidGhostTask(isGhostTask), MESSAGE_CONSTRAINTS);
-        this.isGhostTask = Boolean.parseBoolean(isGhostTask);
+    @Override
+    public GhostTask markAsDone() {
+        String overdueStatus = isTaskOverdue() ? "true" : "false";
+        return new GhostTask(this.getDesc(), this.getDateTime(), super.getRelatedNames(),
+                new Status("true", overdueStatus), super.getRecurrence());
     }
 
     /**
-     * Returns true if a given string is a valid GhostTask.
+     * Marks task as overdue.
+     *
+     * @return same task object that has been marked as overdue
      */
-    public static boolean isValidGhostTask(String test) {
-        return test.matches(VALIDATION_REGEX);
+    @Override
+    public GhostTask markAsOverdue() {
+        String completedStatus = isTaskDone() ? "true" : "false";
+        return new GhostTask(this.getDesc(), this.getDateTime(), super.getRelatedNames(),
+                new Status(completedStatus, "true"), super.getRecurrence());
     }
 
-    public boolean checkIfGhostTask() {
-        return this.isGhostTask;
-    }
 
+    /**
+     * Resets the overdue status of the task.
+     *
+     * @return same task object that has been marked as undone and not overdue
+     */
     @Override
-    public String toString() {
-        return Boolean.toString(isGhostTask);
+    public GhostTask markAsNotOverdue() {
+        String completedStatus = isTaskDone() ? "true" : "false";
+        return new GhostTask(this.getDesc(), this.getDateTime(), super.getRelatedNames(),
+                new Status(completedStatus, "false"), super.getRecurrence());
     }
 
+    /**
+     * Updates the date of the recurring task such that it is not overdue.
+     *
+     * @return same task object that has a date in the future
+     */
     @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof GhostTask // instanceof handles nulls
-                && isGhostTask == ((GhostTask) other).isGhostTask); // state check
+    public GhostTask updateDateRecurringTask() {
```

Comment:
Similarly, it seems that this method should not be called for a ghost task. Perhaps an exception can be thrown and the other 2 private methods: `changeTaskDate` and `changeTaskDateBasedOnRecurrence` can be removed.

Reply:
Actually my initial implementation was indeed to throw an `AssertionError` with "this method should not be called". However, I was worried that it might break LSP. What do you think?